### PR TITLE
datagridfield pattern

### PR DESCRIPTION
**Pull request by thet**
To improve the current unstable javascript situation this PR rebuilds the JS functionality within a pattern.
This provides a defined scope to operate on and should give automatic initialization within modals, mosaic and so on.

..Upgrade notice::
  In this version a pattern handles all the JavaScript functionality to ensure best possible encapsulation between different instances of the datagridfield widget.
  If you have customized templates, make sure you do the following:
  - Add the class ``pat-datagridfield`` to the ``datagridwidget-table-view`` nodes (files: datagridfield_input.pt, datagridfield_input_block.pt).
  - Add the following classes instead of ids to the ui buttons as shown in this map (files: datagridfieldobject_input.pt, datagridfieldobject_input_block.pt_):
    - #btn-addrow -> .dgf--row-add
    - #btn-deleterow -> .dgf--row-delete
    - #btn-moveup -> .dgf--row-moveup
    - #btn-movedown -> .dgf--row-movedown

- Clean up upgrade profile definitions and align to Plone standards.

- Remove non-unique id attributes from UI buttons and add classes instead.
  Let the button functionality be initialized by the pattern.

- Remove ``extra`` parameter for datagridfield widget as it was unused and untested.

- Use pat-datagridfield in templates.

- Rework JavaScript as a Pattern for better initialization.

This targets Plone 5.1 and Plone 5.2, as master does.
A quick screenshot with some rows and enabled/disabled buttons. The last row have all buttons disabled as this is a auto-append row - it will automatically add another row if a input-focusout event happens in one of the auto append row fields. Here the buttons do not make any sense.
Screenshot from 2021-03-05 14-08-57

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,8 +1,43 @@
...
+2.0 (unreleased)
+----------------
+
+- Register new pat-datagridfield bundle.
+
+  Breaking change:
+  The JavaScript resources have changed a lot.
+  If you were customizing templates which loaded these JavaScript resources
+  or customized the JavaScript functionality itself, take special care.
+  The bundle is loaded only for anonymous users.
```
Comment (jensens) [inline CHANGES.rst:13]: only anonymous? or vice versa?

**Review by jensens — APPROVED** [review]
Looks good to me, I would release a beta of 2.0 first and let more people use it in the wild.
For Plone 6 this would need other classes applied to the HTML, right?

File: CHANGES.rst (hunk 1 above)
Comment (thet) [reply CHANGES.rst:13]: vice versa - will fix wording!

**Comment by thet** [comment]
@jensens  tnx for the review! will merge.

**Pull request merged**